skbio/metadata/_interval.py

```diff
@@ -90,6 +91,10 @@ def __init__(self, interval_metadata, locations,
 
     def _add(self):
         """Add the current ``Interval`` to the IntervalMetadata object."""
+        if self.locations[-1][-1] > self._interval_metadata.upper_bound:
```

Comment:
not adding `lower_bound` as a param to the `IntervalMetadata.__init__`, right? I really hesitate to expose it in the constructor for users to set it.
```
def __init__(self, upper_bound):
     self._lower_bound = 0
     ...
```